xclim/indices/_hydrology.py

```diff
@@ -279,3 +283,115 @@ def melt_and_precip_max(
     out = agg.resample(time=freq).max(dim="time")
     out.attrs["units"] = snw.units
     return out
+
+
+@declare_units(q="[discharge]")
+def flow_index(q: xr.DataArray, p: float = 0.95, **indexer) -> xr.DataArray:
+    """
+    Flow index
+
+    Calculate the pth percentile of daily streamflow normalized by the median flow.
+
+    Parameters
+    ----------
+    q : xr.DataArray
+        Daily streamflow data.
+    p : float
+        Percentile for calculating the flow index, between 0 and 1. Default of 0.95 is for high flows.
+    indexer
+        Indexing parameters to perform a temporal subset of the data.
+        It accepts the same arguments as :py:func:`xclim.indices.generic.select_time`.
+
+    Returns
+    -------
+    xr.DataArray
+        Normalized Qp, which is the p th percentile of daily streamflow normalized by the median flow.
+
+    References
+    ----------
+    :cite:cts:`addor2018,Clausen2000`
+    """
+    qt = select_time(q, **indexer)
+    qp = qt.quantile(p, dim="time")
+    q_median = qt.median(dim="time")
+    out = qp / q_median
+    out.attrs["units"] = "1"
+    return out
+
+
+@declare_units(q="[discharge]")
+def high_flow_frequency(
+    q: xr.DataArray, threshold_factor: int = 9, freq: str = "YS-OCT", **indexer
+) -> xr.DataArray:
+    """
+    High flow frequency.
+
+    Calculate the number of days in a given period with flows greater than a specified threshold, given as a
+    multiple of the median flow. By default, the period is the water year starting on 1st October and ending on
+    30th September, as commonly defined in North America.
+
+    Parameters
+    ----------
+    q : xr.DataArray
+        Daily streamflow data.
+    threshold_factor : int
+        Factor by which the median flow is multiplied to set the high flow threshold, default is 9.
+    freq : str, optional
+        Resampling frequency, default is 'YS-OCT' for water year starting in October and ending in September.
+    indexer
+        Indexing parameters to perform a temporal subset of the data.
```

Comment:
```suggestion
        Indexing parameters to perform a temporal subset of the data. The median flow is computed over the whole period, but the number of days only on the subset.
```

Reply:
The median vs mean part is ok, that's how it's done in the reference paper. 

For the subset, that's my mistake. I thought it would make more sense to compute the median and mean with respect to the entire year, but I think it would clash with how indexing is done elsewhere.